### PR TITLE
Remove purify-ts library

### PR DESCRIPTION
**Pull request by franciscog-kainos**
This PR removes `purify-ts` dependency and cleans up API of session class.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
     namespace Express {
         export interface Request {
-            session: Maybe<Session>;
+            session?: Session;
         }
     }
 }
-export { Session, VerifiedSession, ISession, SessionStore, SessionMiddleware, CookieConfig, EitherUtils, Maybe, Either};
+
+export { Session, ISession, SessionStore, SessionMiddleware, CookieConfig};
```
Comment (akaJSaunders) [inline src/index.ts:16]:
```suggestion
export { Session, ISession, SessionStore, SessionMiddleware, CookieConfig };
```

File: src/session/store/SessionStore.ts [hunk 2]
```diff
@@ -21,65 +14,55 @@ export class SessionStore {
...
 
-    public set = (key: string, value: string, timeToLiveInSeconds: number): EitherAsync<Failure, string> => {
-
-        const promise = this.client.set(key, value, "EX", timeToLiveInSeconds)
-            .then(r => Right(r))
-            .catch(err => Left(Failure(StoringError(err, key, value))));
+    public set = async (key: string, value: string, timeToLiveInSeconds: number): Promise<void> => {
 
-        return wrapPromise<Failure, string>(promise);
+        return this.client.set(key, value, "EX", timeToLiveInSeconds)
```
Comment (akaJSaunders) [inline src/session/store/SessionStore.ts:37]: what is "EX" supposed to mean? maybe create a const to store this value for readability?

File: src/session/model/Session.ts [hunk 3]
```diff
@@ -1,109 +1,52 @@
...
-            return Left(
-                Failure(ExpiresMissingError)
-            );
+            throw new IncompleteSessionDataError(SessionKey.Expires);
         }
-        // This time corresponds to the time precisison given by the accounts service in seconds.
-        const dateNowMillis = Number(Date.now().toPrecision(10)) / 1000;
 
+        // This time corresponds to the time precision given by the accounts service in seconds.
+        const dateNowMillis = Number(Date.now().toPrecision(10)) / 1000;
```
Comment (akaJSaunders) [inline src/session/model/Session.ts:49]: in the SessionErrors.ts, you spelled out Milliseconds in full

```suggestion
        const dateNowMilliseconds = Number(Date.now().toPrecision(10)) / 1000;
```

File: test/middleware/session.middleware.test.ts [hunk 4]
```diff
@@ -52,38 +50,38 @@ describe("Session Middleware", () => {
...
         it("should load a session and insert the session object in the request", async () => {
             const sessionStore = Substitute.for<SessionStore>();
-            sessionStore.load(cookieArg()).returns(wrapEither(Either.of(session.data)));
+            sessionStore.load(cookieArg()).returns(Promise.resolve(sessionData));
 
             await SessionMiddleware(config, sessionStore)(request, Substitute.for<express.Response>(), nextFunction);
 
-            expect(request.session.isJust()).to.equal(true);
-            expect(request.session.extract().data).to.be.deep.equal(session.data);
+            // @ts-ignore
```
Comment (akaJSaunders) [inline test/middleware/session.middleware.test.ts]: do we need ts-ignore here?

File: src/session/store/SessionStore.ts (hunk 2 above)
Comment (geek-soft) [reply src/session/store/SessionStore.ts:37]: EX defines expiry mode for Redis client. It is so buried (wrapper class is not exposed) and so insignificant that I do not think adding constant provides much value here.

File: test/middleware/session.middleware.test.ts (hunk 4 above)
Comment (geek-soft) [reply test/middleware/session.middleware.test.ts]: Already removed but historically it was supported to give access to internal sessions data. In the end I decided to restore public access to session data.

File: test/store/store.test.ts [hunk 5]
```diff
@@ -23,21 +24,22 @@ describe("Store", () => {
...
-                _ => assert.fail(),
-                response => expect(response).to.be.deep.equal(data)
-            );
+            expect(await new SessionStore(redis).load(cookie)).to.be.deep.equal(data);
 
             redis.received().get(cookie.sessionId);
         });
 
-        it("should return failure when read failed", async () => {
+        it("should throw error when read failed",  async () => {
```
Comment (akaJSaunders) [inline test/store/store.test.ts:32]:
```suggestion
        it("should throw error when read failed", async () => {
```

**Pull request merged**